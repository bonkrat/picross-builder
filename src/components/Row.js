import React from "react";
import styled from "styled-components";
import generateClues from "../utils/generateClues";

export const Row = ({ className, width, selectedCells, children }) => (
  <div className={className}>
    <CellGroupValues width={width}>
      {generateClues(selectedCells).map((num, index) => (
        <CellGroupValue key={index}>{num}</CellGroupValue>
      ))}
    </CellGroupValues>
    <Cells>{children}</Cells>
  </div>
);

const CellGroupValue = styled.div`
  padding: 0 10px;
  font-weight: bold;
`;

const CellGroupValues = styled.div`
  width: 200px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 0 20px;
`;

const Cells = styled.div`
  display: flex;
  justify-content: center;
`;

const StyledRow = styled(Row)`
  display: flex;
  justify-content: flex-end;
`;

StyledRow.defaultProps = {
  selectedCells: []
};

export default StyledRow;
