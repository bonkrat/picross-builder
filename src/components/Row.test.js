import React from "react";
import { Row } from "./Row";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("Row", () => {
  it("displays the groups of selected cells", () => {
    const wrapper = shallow(
      <Row className="foo" selectedCells={[8, 5, 4, 1]}>
        <div>foobar</div>
      </Row>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
