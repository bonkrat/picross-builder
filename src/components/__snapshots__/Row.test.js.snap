// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row displays the groups of selected cells 1`] = `ShallowWrapper {}`;
